janusgraph-cql/src/main/java/org/janusgraph/diskstorage/cql/CQLKeyColumnValueStore.java

```diff
@@ -117,9 +121,49 @@ public class CQLKeyColumnValueStore implements KeyColumnValueStore {
     public static final String KEY_END_BINDING = "keyEnd";
     public static final String LIMIT_BINDING = "maxRows";
 
-    public static final Function<? super Throwable, BackendException> EXCEPTION_MAPPER = cause -> Match(cause).of(
+    /**
+     * CQL keys are limited to 64k, and a query that attempts to filter on such
+     * a key with a value that exceeds that length will produce a ServerError
+     * on ScyllaDB, perhaps due to a bug, rather than an InvalidQueryException.
+     * This type of ServerError is a permanent failure, since the same query with
+     * the same value will always produce this error. Properly treating this as a
+     * permanent failure avoids excessive retrying of bad CQL queries and allows
+     * upstream code to properly report the exception.
+     */
+    private static boolean isKeySizeTooLarge(Throwable cause) {
+        AllNodesFailedException failed = ExceptionUtils.throwableOfType(cause, AllNodesFailedException.class);
+        if (failed == null) {
+            return false;
+        }
+        return failed.getAllErrors().entrySet().stream().anyMatch(entry -> {
+            return entry.getValue().stream().anyMatch(throwable ->  {
+                if (throwable instanceof ServerError) {
+                    ServerError error = (ServerError) throwable;
+                    if (error.getMessage().startsWith("Key size too large:")) {
+                        return true;
+                    }
+                    return false;
+                }
+                return false;
+            });
+        });
+    }
```

Comment:
As Codacy commented in this PR.
1) it's better to declare all methods after all fields are declared. I.e. it's better to move this method down after all fields are declared.
2) It's possible to make boolean statements easier without if / else statements. I.e. something like below:
```
private static boolean isKeySizeTooLarge(Throwable cause) {
    AllNodesFailedException failed = ExceptionUtils.throwableOfType(cause, AllNodesFailedException.class);
    if (failed == null) {
        return false;
    }
    return failed.getAllErrors().values().stream().anyMatch(errors ->
        errors.stream().anyMatch(error -> (error instanceof ServerError)
            && error.getMessage().startsWith("Key size too large:")));
}
```